### PR TITLE
Initialize

### PR DESCRIPTION
**Pull request by sarkis**
## what
Terraform module for creating an IAM user and access id/secret with access to SSM+KMS secrets.

## why
Make it easier for SaaS CI/CD solutions to access secrets from SSM+KMS.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2017-2018 Cloud Posse, LLC
```
Comment (osterman) [inline LICENSE]: Fix years

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		actions = [
+		]
+
+		resources = ["*"]
+	}
+}
+
+variable "policy" {
+  description = "User policy in json format"
+  default     = "${data.aws_iam_policy_document.default.json}"
```
Comment (osterman) [inline variables.tf]: Interpolations are not valid as defaults

**Comment by sarkis** [comment]
```
Terraform will perform the following actions:

  + module.chamber_iam_user.module.chamber_user.aws_iam_access_key.default
      id:                  <computed>
      encrypted_secret:    <computed>
      key_fingerprint:     <computed>
      secret:              <computed>
      ses_smtp_password:   <computed>
      status:              <computed>
      user:                "ps-dev-chamber"

  + module.chamber_iam_user.module.chamber_user.aws_iam_user.default
      id:                  <computed>
      arn:                 <computed>
      force_destroy:       "false"
      name:                "ps-dev-chamber"
      path:                "/"
      unique_id:           <computed>

  + module.chamber_iam_user.module.chamber_user.aws_iam_user_policy.default
      id:                  <computed>
      name:                "ps-dev-chamber"
      policy:              "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Sid\": \"\",\n      \"Effect\": \"Allow\",\n      \"Action\": [\n        \"ssm:GetParameters\",\n        \"ssm:DescribeParameters\"\n      ],\n      \"Resource\": \"*\"\n    },\n    {\n      \"Sid\": \"\",\n      \"Effect\": \"Allow\",\n      \"Action\": \"kms:Decrypt\",\n      \"Resource\": \"arn:aws:kms:us-west-2:010570371725:key/365b6e64-5c29-4460-bdf2-6b857d5e71f0\"\n    }\n  ]\n}"
      user:                "ps-dev-chamber"

  + module.chamber_iam_user.module.label.null_resource.default
      id:                  <computed>
      triggers.%:          "5"
      triggers.attributes: ""
      triggers.id:         "ps-dev-chamber"
      triggers.name:       "chamber"
      triggers.namespace:  "ps"
      triggers.stage:      "dev"

  + module.chamber_iam_user.module.chamber_user.module.label.null_resource.default
      id:                  <computed>
      triggers.%:          "5"
      triggers.attributes: ""
      triggers.id:         "ps-dev-chamber"
      triggers.name:       "chamber"
      triggers.namespace:  "ps"
      triggers.stage:      "dev"


Plan: 5 to add, 0 to change, 0 to destroy.
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    packages:
+      - git
+      - make
+      - curl
+
+install:
+  - make init
+
+script:
+  - make terraform:install
```
Comment (osterman) [inline .travis.yml]: replace `:` with `/`

File: README.md [hunk 4]
```diff
@@ -1 +1,142 @@
-# terraform-aws-iam-chamber-user
\ No newline at end of file
+# terraform-aws-iam-chamber-user [![Build Status](https://travis-ci.org/cloudposse/terraform-aws-iam-chamber-user.svg)](https://travis-ci.org/cloudposse/terraform-aws-iam-chamber-user)
+
+Terraform Module to provision a basic IAM system user with access to SSM parameters, suitable for CI/CD Systems
```
Comment (osterman) [inline README.md]: `s/system/chamber/`

Comment (osterman) [reply README.md]: Link chamber to https://github.com/segmentio/chamber

File: README.md [hunk 5]
```diff
@@ -1 +1,142 @@
...
+
+## Usage
+
+### Simple usage
+
+```hcl
+module "circleci" {
+  source     = "git::https://github.com/cloudposse/terraform-aws-iam-chamber-user.git?ref=master"
+  namespace  = "cp"
+  stage      = "circleci"
```
Comment (osterman) [inline README.md]: let's rename to `staging`

File: README.md [hunk 6]
```diff
@@ -1 +1,142 @@
...
+}
+```
+
+### Full usage
+
+```hcl
+module "circleci" {
+  source      = "git::https://github.com/cloudposse/terraform-aws-iam-chamber-user.git?ref=master"
+  namespace   = "cp"
+  stage       = "circleci"
```
Comment (osterman) [inline README.md]: lets rename to `staging`

File: README.md [hunk 7]
```diff
@@ -1 +1,142 @@
...
+
+### Contributors
+
+| [![Erik Osterman][erik_img]][erik_web]<br/>[Erik Osterman][erik_web] | [![Andriy Knysh][andriy_img]][andriy_web]<br/>[Andriy Knysh][andriy_web] |
+|-------------------------------------------------------|------------------------------------------------------------------|
+
+  [erik_img]: http://s.gravatar.com/avatar/88c480d4f73b813904e00a5695a454cb?s=144
+  [erik_web]: https://github.com/osterman/
+  [andriy_img]: https://avatars0.githubusercontent.com/u/7356997?v=4&u=ed9ce1c9151d552d985bdf5546772e14ef7ab617&s=144
+  [andriy_web]: https://github.com/aknysh/
```
Comment (osterman) [inline README.md]: You can add yourself

**Review by osterman — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -1 +1,142 @@
...
+
+Terraform Module to provision a basic IAM system user with access to SSM parameters, suitable for CI/CD Systems
+(_e.g._ TravisCI, CircleCI, CodeFresh) or systems which are *external* to AWS that cannot leverage [AWS IAM Instance Profiles](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html).
+
+We do not recommend creating IAM users this way for any other purpose.
+
+
+## Usage
+
+### Simple usage
```
Comment (osterman) [inline README.md]: We don't need 2 usages for this module. Only a single one.

File: README.md [hunk 9]
```diff
@@ -1 +1,142 @@
...
+
+### Full usage
+
+```hcl
+module "circleci" {
+  source      = "git::https://github.com/cloudposse/terraform-aws-iam-chamber-user.git?ref=master"
+  namespace   = "cp"
+  stage       = "circleci"
+  name        = "secrets"
+  kms_key_arn = ""
```
Comment (osterman) [inline README.md]: Add a mock kms key

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    ]
+  }
+}
+
+module "chamber_user" {
+  source    = "git::https://github.com/cloudposse/terraform-aws-iam-system-user.git?ref=master"
+  name      = "${var.name}"
+  enabled   = "${var.enabled}"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
```
Comment (osterman) [inline main.tf]: missing attributes

File: main.tf [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+data "aws_iam_policy_document" "default" {
+  statement {
+    actions = [
+      "ssm:DescribeParameters",
+      "ssm:GetParameters",
+    ]
+
+    resources = ["*"]
```
Comment (osterman) [inline main.tf]: move this to a variable `ssm_resources` with a default value of `["*"]`

File: main.tf [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  name       = "${var.name}"
+  attributes = "${var.attributes}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+  enabled    = "${var.enabled}"
+}
+
+data "aws_iam_policy_document" "default" {
+  statement {
+    actions = [
```
Comment (osterman) [inline main.tf]: Move this to a variable named `ssm_actions` with a default value of what you have here.

File: .travis.yml (hunk 3 above)
Comment (osterman) [reply .travis.yml]: ...for all targets

Comment (sarkis) [reply .travis.yml]: :+1:

File: README.md [hunk 13]
```diff
@@ -1 +1,124 @@
...
+## Usage
+
+### Simple usage
+
+```hcl
+module "circleci" {
+  source     = "git::https://github.com/cloudposse/terraform-aws-iam-chamber-user.git?ref=master"
+  namespace  = "cp"
+  stage      = "staging"
+  name       = "chamber"
```
Comment (osterman) [inline README.md]: this will definitely need the `kms_key_arn`

File: README.md [hunk 14]
```diff
@@ -1 +1,124 @@
...
+| `name`          |   ``    | Name  (e.g. `bastion` or `db`)                                                              |   Yes    |
+| `attributes`    |  `[]`   | Additional attributes (e.g. `policy` or `role`)                                             |    No    |
+| `tags`          |  `{}`   | Additional tags  (e.g. `map("BusinessUnit","XYZ")`                                          |    No    |
+| `delimiter`     |   `-`   | Delimiter to be used between `name`, `namespace`, `stage`, `arguments`, etc.                |    No    |
+| `force_destroy` | `false` | Destroy even if it has non-Terraform-managed IAM access keys, login profile or MFA devices. |    No    |
+| `path`          |   `/`   | Path in which to create the user                                                            |    No    |
+| `enabled`       | `true`  | Set to `false` to prevent the module from creating any resources                            |    No    |
+| `kms_key_arn`   |   ``    | KMS key_arn used if Secure Strings are stored in Parameter Store to decrypt secrets.        |    No    |
```
Comment (osterman) [inline README.md]: This should be required

File: variables.tf [hunk 15]
```diff
@@ -0,0 +1,50 @@
+variable "name" {}
```
Comment (osterman) [inline variables.tf]: Use the same variables we have in all of our other modules.

Comment (osterman) [reply variables.tf]: https://github.com/cloudposse/terraform-aws-key-pair/blob/master/variables.tf#L1-L32

They are all well documented

File: variables.tf [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+variable "stage" {
+  default = ""
+}
+
+variable "attributes" {
+  type    = "list"
+  default = []
+}
+
+variable "ssm_actions" {
```
Comment (osterman) [inline variables.tf]: Add `description` to all variables.

File: README.md (hunk 13 above)
Comment (osterman) [reply README.md]: Please add mock `kms_key_arn`

File: README.md [hunk 17]
```diff
@@ -1 +1,125 @@
...
+
+### Simple usage
+
+```hcl
+module "circleci" {
+  source     = "git::https://github.com/cloudposse/terraform-aws-iam-chamber-user.git?ref=master"
+  namespace   = "cp"
+  stage       = "staging"
+  name        = "chamber"
+  kms_key_arn = "arn:aws:kms:region:account-id:key/CMK"
```
Comment (osterman) [inline README.md]: Add a realistic arn. It's hard to know what is a placeholder vs what is replaced.

**Review by osterman — APPROVED** [review]

**Pull request merged**